code/modules/food_and_drinks/drinks/drinks.dm

```diff
@@ -22,22 +22,29 @@
 		gulp_size = max(round(reagents.total_volume / 5), 5)
 
 /obj/item/reagent_containers/food/drinks/attack(mob/living/M, mob/user, def_zone)
-
 	if(!reagents || !reagents.total_volume)
-		to_chat(user, "<span class='warning'>[src] is empty!</span>")
-		return 0
+		if(iscyborg(user))
+			var/mob/living/silicon/robot/R = user
+			R.module.synthetize_reagents() //maybe this will work?
+		else
+			to_chat(user, "<span class='warning'>[src] is empty!</span>")
+		return FALSE
 
 	if(!canconsume(M, user))
-		return 0
+		return FALSE
 
 	if (!is_drainable())
 		to_chat(user, "<span class='warning'>[src]'s lid hasn't been opened!</span>")
-		return 0
+		return FALSE
 
-	if(M == user)
+	if(M == user) //Because M doesn't include silicons I guess?
 		user.visible_message("<span class='notice'>[user] swallows a gulp of [src].</span>", "<span class='notice'>You swallow a gulp of [src].</span>")
 		if(M.has_trait(TRAIT_VORACIOUS))
 			M.changeNext_move(CLICK_CD_MELEE * 0.5) //chug! chug! chug!
+	else if(iscyborg(user))
+		var/mob/living/silicon/robot/R = user
+		R.module.analyze_reagents(reagents.reagent_list) //Analyze everything inside the container
+		reagents.clear_reagents() //Consumes everything
```

Comment:
Has any of this proc been tested post-changes?

Reply:
Nope. And it is all web pr too.